SwiftGit2/Repository.swift

```diff
@@ -537,4 +537,149 @@ final public class Repository {
 		let iterator = CommitIterator(repo: self, root: branch.oid.oid)
 		return iterator
 	}
+
+	// MARK: - Diffs
+
+	public func diff(for commit: Commit) -> Result<[Diff.Delta], NSError> {
+		return self
+			.commit(with: commit.oid)
+			.flatMap { baseCommit in
+				return self.tree(from: baseCommit)
+			}
+			.flatMap { baseTree in
+				guard !commit.parents.isEmpty else {
+					// Initial commit in a repository
+					return self.diff(withOldTree: nil, andNewTree: baseTree)
+				}
+
+				var mergeDiff: Result<OpaquePointer?, NSError> = .success(nil)
+				for parent in commit.parents {
+					mergeDiff = mergeDiff
+						.flatMap { mergeDiff in
+							return self
+								.parentCommit(from: parent)
+								.flatMap { commit in
+									return self.tree(from: commit)
+								}
+								.flatMap { tree in
+									return self.diff(withOldTree: tree, andNewTree: baseTree)
+								}
+								.flatMap { diff in
+									guard let mergeDiff = mergeDiff else { return .success(diff) }
+									let mergeResult = git_diff_merge(mergeDiff, diff)
+									guard mergeResult == GIT_OK.rawValue else {
+										return .failure(NSError(gitError: mergeResult, pointOfFailure: "git_diff_merge"))
+									}
+									return .success(mergeDiff)
+							}
+					}
+				}
+				return .success(mergeDiff.value!!)
```

Comment:
I think we can safely get rid of this `!!`.

```swift
			.flatMap { baseTree in
				var mergeDiff: Result<OpaquePointer?, NSError> = .success(nil)
				for parent in commit.parents {
					mergeDiff = mergeDiff
						.flatMap { mergeDiff in
							return self
								.parentCommit(from: parent)
								.flatMap { commit in
									return self.tree(from: commit)
								}
								.flatMap { tree in
									return self.diff(withOldTree: tree, andNewTree: baseTree)
								}
								.flatMap { diff in
									guard let mergeDiff = mergeDiff else { return .success(diff) }
									let mergeResult = git_diff_merge(mergeDiff, diff)
									guard mergeResult == GIT_OK.rawValue else {
										return .failure(NSError(gitError: mergeResult, pointOfFailure: "git_diff_merge"))
									}
									return .success(mergeDiff)
							}
					}
				}
				return mergeDiff.map { ($0, baseTree) }
		        }
			.flatMap { (diffResult, baseTree) in
				if let diffResult = diffResult {
				    return self.processDiffDeltas(diffResult)
				} else {
				    // Initial commit in a repository
				    return self.diff(withOldTree: nil, andNewTree: baseTree)
				}
		}
```

Reply:
Removing the !! gives me a strange error I don't understand: `Member 'success' in 'Result<OpaquePointer, NSError>' produces result of type 'Result<T, Error>', but context expects 'Result<OpaquePointer, NSError>'`